src/coreclr/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreCLR.cs

```diff
@@ -234,6 +234,9 @@ private static unsafe bool InitializeConfigAndDetermineUsePortableThreadPool()
                 ? AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.EnableWorkerTracking", false)
                 : GetEnableWorkerTrackingNative();
 
+        internal static readonly bool EnableDispatchAutoreleasePool =
+            AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.EnableDispatchAutoreleasePool", false);
```

Comment:
Do we expect these two AppContext switches to always default to false? The current naming pattern we've used is to use the pattern EnableXXX when XXX is off by default (or will be off by default in the future - like with BinaryFormatter).

My naive understanding from looking at the code below is this switch only affects MacOS. If you are on any platform other than Apple, this switch doesn't do anything. If you are on an Apple OS that isn't MacOS, the behavior will be "on".

Can you also update https://github.com/dotnet/runtime/blob/master/docs/workflow/trimming/feature-switches.md with your new feature switch?

Reply:
> My naive understanding from looking at the code below is this switch only affects MacOS.

Yep, there is a bit of efficiency that can be gained if we put the reading of the switch under old-fashined ifdef. As it is written, we will read the switch even on platforms where it does nothing.

Reply:
I'm questioning if we need a full-blown "feature switch" for this. Not all AppContext switches need to follow the feature switch design (added ILLink.Substitutions.xml entry, and MSBuild entry). If this only affects MacOS, we aren't really concerned about trimming out every bit of code on MacOS.

Reply:
@lambdageek specifically requested that we add a linker substitution for it.

Reply:
I guess I don't fully understand the value. Adding and maintaining these feature switches isn't "free". They are kind of like public API that we need to document and maintain compatibility.

The only thing I'm seeing this enable is if you wanted to trim an `osx` self contained application, you could set this switch to trim this code. Is it a lot of code being trimmed? Size of `osx` apps aren't really a major concern. I guess I just don't see the value outweighing the cost here.

cc @stephentoub @vitek-karas @marek-safar as well

Reply:
I think this should be a compile-time constant for RIDs where `false` value should never be set (e.g. iOS). 

This switch changes the behavior so I think we should be consistent on all Apple platforms or the developers will be surprised that their library works on iOS and does not on macOS.

Reply:
> I think this should be a compile-time constant for RIDs where `false` value should never be set (e.g. iOS).

Customers are able to turn this off in Xamarin.iOS/Xamarin.Mac, so I think they should continue to be able to do so in .NET 6 if they so wish.

Reply:
I'll change the condition to use the feature flag on the mobile Apple platforms as well.

Now the question is what should the default be?

Reply:
I believe the existing default is true /cc @rolfbjarne

Reply:
The existing default is `true` (and that's what it should be too in .NET 6).

Reply:
The existing default on CoreCLR is false.

Reply:
How does the following sound to everyone? @rolfbjarne @jkotas 

The default for in the runtime for when the flag is not specified is `false`. So, for the net6.0 TFM without a target platform, the default is `false`.

The MSBuild SDK tooling for `net6.0-macos`, `net6.0-ios`, etc. sets the feature flag default to `true` in MSBuild. That way people migrating to `net6.0-*` from the Xamarin world to the TFMs with target platform APIs will continue to have the same default.

Reply:
Sounds reasonable to me.